src/utils/metrics-calculator.ts

```
@@ -9,163 +9,103 @@ export class MetricsCalculator {
       .map(line => line.trim())
       .filter(line => line.length > 0 && !line.match(/^\s*[{}]\s*$/));
     
+    // Combine all metrics calculation in a single AST traversal
+    const combinedMetrics = this.calculateAllMetricsInSinglePass(node);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)